src/main/java/com/onlinebookstore/onlinebookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,148 @@
+                        "Shopping cart not found for user id: " + user.getId()));
+
+        Order order = orderMapper.toModel(orderRequestDto);
+        order.setUser(user);
+        order.setOrderStatus(OrderStatus.PENDING);
+        order.setOrderItems(cart.getCartItems().stream()
+                .map(cartItem -> {
+                    OrderItem orderItem = new OrderItem();
+                    orderItem.setOrder(order);
+                    orderItem.setBook(cartItem.getBook());
+                    orderItem.setQuantity(cartItem.getQuantity());
+                    orderItem.setPrice(cartItem.getBook().getPrice());
+                    return orderItem;
+                }).collect(Collectors.toSet()));
+        order.setTotal(cart.getCartItems().stream()
+                .map(cartItem -> cartItem.getBook()
+                        .getPrice()
+                        .multiply(BigDecimal
+                                .valueOf(cartItem
+                                        .getQuantity())))
+                .reduce(BigDecimal.ZERO, BigDecimal::add));
+        order.setOrderDate(LocalDateTime.now());
+        order.setShippingAddress(orderRequestDto.getShippingAddress());
+
+        Order savedOrder = orderRepository.save(order);
+
+        cart.getCartItems().forEach(cartItem -> shoppingCartService
+                .removeBookFromCart(cartItem.getId(), user.getId()));
+        shoppingCartRepository.save(cart);
+
+        return orderMapper.toDto(savedOrder);
+    }
+
+    @Override
+    public Set<OrderResponseDto> getOrderHistory(User user) {
+        Set<Order> orders = orderRepository.findByUser(user);
+        if (orders.isEmpty()) {
+            throw new EntityNotFoundException("No orders found for user id: "
+                    + user.getId());
+        }
+        return orders.stream()
+                .map(orderMapper::toDto)
+                .collect(Collectors.toSet());
+    }
+
+    @Override
+    public OrderResponseDto updateOrderStatus(Long orderId,
+                                              OrderUpdateStatusDto orderUpdateStatusDto) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(() -> new EntityNotFoundException("Order not found with id: "
+                        + orderId));
+        System.out.println("Updating order status to: "
+                + orderUpdateStatusDto.getOrderStatus());
+
+        order.setOrderStatus(orderUpdateStatusDto.getOrderStatus());
+
+        if (order.getOrderStatus() == OrderStatus.COMPLETED) {
+            ShoppingCart cart = shoppingCartRepository.findByUserId(order.getUser().getId())
+                    .orElseThrow(() -> new EntityNotFoundException(
+                            "Shopping cart not found for user id: "
+                                    + order.getUser().getId()));
+            cart.getCartItems().clear();
+            shoppingCartRepository.save(cart);
+        }

```

Comment:
Why this logic required? You already cleaned shopping cart when order was created

Reply:
Honestly I got some idea about this but right now cannot remeber it... I guess i will delete it then